python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -208,8 +208,15 @@ def run(self):
         """
         try:
             # Run scheduling loop until complete.
-            while self._scheduling_loop_step(self._topology) and not self._shutdown:
-                pass
+            while True:
+                # use process_time to avoid timing ray.wait in _scheduling_loop_step
+                t_start = time.process_time()
+                continue_sched = self._scheduling_loop_step(self._topology)
+                self._initial_stats.streaming_exec_schedule_s.add(
+                    time.process_time() - t_start
+                )
+                if not continue_sched or self._shutdown:
+                    break
```

Comment:
nit, better to use the context manager to make code cleaner and deal with exceptions

Reply:
Oh yeah much cleaner, thanks!